File: src/main/java/cz/vhromada/catalog/web/fo/GenreFO.java
package cz.vhromada.catalog.web.fo;

import java.io.Serializable;
import java.util.Objects;

import javax.validation.constraints.NotBlank;

/**
 * A class represents FO for genre.
 *
 * @author Vladimir Hromada
 */
public class GenreFO implements Serializable {

    /**
     * SerialVersionUID
     */
    private static final long serialVersionUID = 1L;

    /**
     * ID
     */
    private Integer id;

    /**
     * Name
     */
    private @NotBlank String name;

    /**
     * Position
     */
    private int position;

    /**
     * Returns ID.
     *
     * @return ID
     */
    public Integer getId() {
        return id;
    }

    /**
     * Sets a new value to ID.
     *
     * @param id new value
     */
    public void setId(final Integer id) {
        this.id = id;
    }

    /**
     * Returns name.
     *
     * @return name
     */

    public String getName() {
        return name;
    }

    /**
     * Sets a new value to name.
     *
     * @param name new value
     */
    public void setName(final String name) {
        this.name = name;
    }

    /**
     * Returns position.
     *
     * @return position
     */
    public int getPosition() {
        return position;
    }

    /**
     * Sets a new value to position.
     *
     * @param position new value
     */
    public void setPosition(final int position) {
        this.position = position;
    }

    @Override
    public boolean equals(final Object obj) {
        if (this == obj) {
            return true;
        }
        if (!(obj instanceof GenreFO) || id == null) {
            return false;
        }

        return id.equals(((GenreFO) obj).id);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(id);
    }

    @Override
    public String toString() {
        return String.format("GenreFO [id=%d, name=%s, position=%d]", id, name, position);
    }

}

File: src/main/kotlin/cz/vhromada/catalog/web/fo/GenreFO.kt
package cz.vhromada.catalog.web.fo

import java.io.Serializable
import java.util.Objects
import javax.validation.constraints.NotBlank

/**
 * A class represents FO for genre.
 *
 * @author Vladimir Hromada
 */
data class GenreFO(

        /**
         * ID
         */
        val id: Int?,

        /**
         * Name
         */
        @field:NotBlank
        val name: String?,

        /**
         * Position
         */
        val position: Int?) : Serializable {

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        return if (other !is GenreFO || id == null) {
            false
        } else id == other.id
    }

    override fun hashCode(): Int {
        return Objects.hashCode(id)
    }

}
